pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go

```diff
@@ -151,9 +152,15 @@ func (m *PerformanceQueryImpl) GetFormattedCounterArrayDouble(hCounter PDH_HCOUN
 		if ret = PdhGetFormattedCounterArrayDouble(hCounter, &buffSize, &itemCount, &buff[0]); ret == ERROR_SUCCESS {
 			items := unsafe.Slice((*PDH_FMT_COUNTERVALUE_ITEM_DOUBLE)(unsafe.Pointer(&buff[0])), itemCount)
 			values := make([]CounterValue, 0, itemCount)
-			for _, item := range items {
+			for i, item := range items {
 				if item.FmtValue.CStatus == PDH_CSTATUS_VALID_DATA || item.FmtValue.CStatus == PDH_CSTATUS_NEW_DATA {
-					val := CounterValue{UTF16PtrToString(item.SzName), item.FmtValue.DoubleValue}
+					var val CounterValue
+					if i == 0 { // Be consistent with perfmon output: instance 0 does not display index.
+						val = CounterValue{UTF16PtrToString(item.SzName), item.FmtValue.DoubleValue}
+					} else {
+						val = CounterValue{fmt.Sprintf("%s_%d", UTF16PtrToString(item.SzName), i), item.FmtValue.DoubleValue}
```

Comment:
If a wildcard matches multiple distinct names (e.g. `[A, A, B, B, B]` then this code will currently label the resulting instances as 
`[A, A_1, B_2, B_3, B_4]`, but it should label them as `[A, A_1, B, B_1, B_2]`.

I know this happens at least for `\Process(*)\% Process Time`, but it can in theory happen with any counter that can have duplicate instance names.